### ISSUE TITLE
Add utility function for lazy loading

### ISSUE THREAD

**Issue by JuliaSprenger** [NeuralEnsemble/python-neo#705]
Similar to the `time_slice` utilities introduced in #526 it would be nice to be able go from a lazy object structure to a structure containing data e.g. via `block.load(time_slice=(...))` or `segment.load(channel_indexes=[1])`. In case the current structure is already non-lazy this should just return a handle to the existing structure.

**Comment by samuelgarcia** [comment]
channel_indexes refer to one AnalogSignal a Segment could have several AnalogSignal and in that case channel_indexes would be ambiguous.

And block.load(time_slice=...) is also ambiguous because:
  * it can cover several segment
  * several segment can reset there time

For this we should have some flag like **global_time_acroos_block** to make it non ambiguous.

**Comment by JuliaSprenger** [comment]
I agree on the block level this might be hard to interpret. However, I think we need some sort of utility functions (maybe only on Segment level then), so users don't have to iterate the whole Neo structure e.g. if they just want to load everything happening during a specific time interval in the recording.

I don't think you need a global time frame for adding a `time_slice` capability to a block since every object has it's own time scale. This function could just work on all segments / data objects present and if they are outside of the requested time interval then they are just ignored.
For channel indexes this will be more complex to interpret, I agree.

**Comment by apdavison** [comment]
Is someone still interested in working on this? @JuliaSprenger @samuelgarcia ?
